src/java.desktop/macosx/classes/com/apple/laf/AquaMenuPainter.java

```diff
@@ -494,13 +494,15 @@ private String layoutMenuItem(final JMenuItem menuItem, final FontMetrics fm, fi
         if (!isTopLevelMenu) {
             //    if ( GetSysDirection() < 0 ) hierRect.right = hierRect.left + w + 4;
             //    else hierRect.left = hierRect.right - w - 4;
-            arrowIconR.x = (viewR.width - arrowIconR.width) + 1;
-            arrowIconR.y = viewR.y + (labelR.height / 2) - (arrowIconR.height / 2) + 1;
 
-            checkIconR.y = viewR.y + (labelR.height / 2) - (checkIconR.height / 2);
+            arrowIconR.x = Math.abs((viewR.width - arrowIconR.width) + 1);
```

Comment:
May I take it that you verified this doesn't have any adverse effects on submenus with text :?

Reply:
Yes, with fix it is
<img width="120" alt="Screenshot 2022-08-08 at 9 46 02 AM" src="https://user-images.githubusercontent.com/43534309/183338104-3df28849-1557-4c65-b22c-2f0db7ec8755.png">

whereas without fix it is
<img width="120" alt="Screenshot 2022-08-08 at 9 47 57 AM" src="https://user-images.githubusercontent.com/43534309/183338213-da66d1d5-c186-4ae2-9b14-4381df4d7dce.png">